configure

```diff
@@ -228,10 +225,15 @@ while true; do
       cudnn_lib_path="${CUDNN_INSTALL_PATH}/lib/libcudnn.dylib"
       cudnn_alt_lib_path="${CUDNN_INSTALL_PATH}/libcudnn.dylib"
     fi
+    # Resolve to the SONAME of the symlink.  Use readlink without -f
+    # to resolve exactly once to the SONAME.  E.g, libcudnn.so ->
+    # libcudnn.so.4.
+    # If the path is not a symlink, readlink will exit with an error code, so
+    # in that case, we return the path itself.
     if [ -f "$cudnn_lib_path" ]; then
-      REALVAL=`readlink -f ${cudnn_lib_path}`
+      REALVAL=`readlink ${cudnn_lib_path} || echo "${cudnn_lib_path}"`
```

Comment:
Looking into this more, I now understand why the comment says that we only want to resolve one level.

On my installation, `libcudnn.so` resolves to `libcudnn.so.4`, which then resolves to `libcudnn.so.4.0.7`, and this code extracts the 4 from `libcudnn.so.4`. Though, the regex that we are using would still work if we resolve all the way to `libcudnn.so.4.0.7`.

In any case, let's merge this since `readlink -f` is invalid on macOS. Once Bazel 0.3.2 (see bazelbuild/bazel#1840) is released, I am planning to move this logic into `cuda_configure` and use the Skylark `realpath()` method, which implements `readlink -f` and works cross-platform.